async-nats/src/lib.rs

```diff
@@ -925,6 +971,12 @@ impl Stream for Subscriber {
     type Item = Message;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        if let Some(max) = self.max {
```

Comment:
One call to poll, doesn't always equal an item.

Reply:
Ah, forgot about difference between Stream and Iter :). Fixed.